### PR TITLE
fix acceptance of one-sided offers

### PR DESCRIPTION
**Pull request by paninaro**
`offerBuildDataToOffer()` now takes an optional boolean parameter that indicates whether the offer column can be empty

### CODE REVIEW

File: packages/gui/src/util/offerBuilderDataToOffer.ts [hunk 1]
```diff
@@ -58,10 +68,12 @@ export default async function offerBuilderDataToOffer(
...
 
-  const hasOffer = !!offeredXch.length || !!offeredTokens.length || !!offeredNfts.length;
+  if (!allowEmptyOfferColumn) {
+    const hasOffer = !!offeredXch.length || !!offeredTokens.length || !!offeredNfts.length;
 
-  if (!hasOffer) {
-    throw new Error(t`Please specify at least one offered asset`);
+    if (!hasOffer) {
+      throw new Error(t`Please specify at least one offered asset`);
+    }
```
Comment (ChiaMineJP) [inline packages/gui/src/util/offerBuilderDataToOffer.ts:76]: I think one-sideded offer includes request-only offer.
How about writing like this?
```typescript
  const hasOffer = !!offeredXch.length || !!offeredTokens.length || !!offeredNfts.length;
  const hasRequest = !!requestedXch.length || !!requestedTokens.length || !!requestedNfts.length;
  if(!hasOffer && !hasRequest){
    throw new Error(t`Offer is empty`);
  }

  if (!allowEmptyOfferColumn) {
    if (!hasOffer || !hasRequest) {
      throw new Error(t`Please specify at least one offering/requesting asset`);
    }
  }
```

File: packages/gui/src/components/offers2/CreateOfferBuilder.tsx [hunk 2]
```diff
@@ -71,13 +71,14 @@ export default function CreateOfferBuilder(props: CreateOfferBuilderProps) {
...
+        data: values,
         wallets,
-        offers || [],
-        false,
-        true
-      );
+        offers: offers || [],
+        validateOnly: false,
+        considerNftRoyalty: true,
+        allowEmptyOfferColumn: false,
```
Comment (ChiaMineJP) [inline packages/gui/src/components/offers2/CreateOfferBuilder.tsx:80]: Is initiating one-sided offer not allowed?

Comment (paninaro) [reply packages/gui/src/components/offers2/CreateOfferBuilder.tsx:80]: Initiating a one-sided offer is allowed when creating a new offer, but that one-sided offer *MUST* be offering some asset. We don't support creating offers that only request assets.

Maybe that will be a feature that we support at some point, but currently the intent is to only support creating an offer if some assets are being offered.

File: packages/gui/src/util/offerBuilderDataToOffer.ts (hunk 1 above)
Comment (paninaro) [reply packages/gui/src/util/offerBuilderDataToOffer.ts:76]: I checked with E-ming, and the intent is to only support one-sided offers where assets are being offered. It isn't currently planned to support a request-only offer.

Comment (ChiaMineJP) [reply packages/gui/src/util/offerBuilderDataToOffer.ts:76]: In that case, the code looks good to me

File: packages/gui/src/components/offers2/CreateOfferBuilder.tsx (hunk 2 above)
Comment (ChiaMineJP) [reply packages/gui/src/components/offers2/CreateOfferBuilder.tsx:80]: Ok, thanks for the detail

**Review by ChiaMineJP — APPROVED** [review]

**Pull request merged**